src/enact/references.py

```diff
@@ -86,6 +89,23 @@ async def commit_async(resource: R) -> 'Ref[R]':
   return await Store.current().commit_async(resource)
 
 
+class _PackHelper:
+  """Collects references and type keys while walking a resource."""
+
+  def __init__(self):
+    self.links: Set[str] = set()
+    self.type_keys: Set[types.TypeKey] = set()
+
+  def __call__(self, value: interfaces.FieldValue):
+    """Collects references and type keys."""
+    if isinstance(value, Ref):
+      self.links.add(value.id)
+    elif isinstance(value, interfaces.ResourceBase):
+      self.type_keys.add(value.type_key())
+    elif isinstance(value, type) and issubclass(value, interfaces.ResourceBase):
+      self.type_keys.add(value.type_key())
+
```

Comment:
nit: handle the `else:` catch all case and assert?

Reply:
It's benign, but I added an explicit else case plus comment to clarify.